internal/fwserver/attr_value.go

```diff
@@ -0,0 +1,178 @@
+package fwserver
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/internal/fwschemadata"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+func coerceListValue(schemaPath path.Path, value attr.Value) (types.List, diag.Diagnostics) {
+	list, ok := value.(types.List)
+
+	if !ok {
+		return types.List{Null: true}, diag.Diagnostics{
+			attributePlanModificationWalkError(schemaPath, value),
+		}
+	}
+
+	return list, nil
+}
+
+func coerceMapValue(schemaPath path.Path, value attr.Value) (types.Map, diag.Diagnostics) {
+	m, ok := value.(types.Map)
+
+	if !ok {
+		return types.Map{Null: true}, diag.Diagnostics{
+			attributePlanModificationWalkError(schemaPath, value),
+		}
+	}
+
+	return m, nil
+}
+
+func coerceObjectValue(schemaPath path.Path, value attr.Value) (types.Object, diag.Diagnostics) {
+	nullObject := types.Object{Null: true}
+
+	object, ok := value.(types.Object)
+
+	if !ok {
+		return nullObject, diag.Diagnostics{
```

Comment:
```suggestion
	object, ok := value.(types.Object)

	if !ok {
		return types.Object{Null: true}, diag.Diagnostics{
```

Tiny nit for consistency.